controllers/idler/idler_controller.go

```diff
@@ -144,8 +144,23 @@ func (r *Reconciler) ensureIdling(logger logr.Logger, idler *toolchainv1alpha1.I
 					}
 					podLogger.Info("Pod deleted")
 				}
+				var appName string
+				var appType string
+				owners := pod.ObjectMeta.OwnerReferences
+				for _, owner := range owners {
+					if owner.Controller != nil && (owner.Kind == "Deployment" || owner.Kind == "ReplicaSet" ||
+						owner.Kind == "StatefulSet" || owner.Kind == "DeploymentConfig" ||
+						owner.Kind == "ReplicationController") {
+						appName = owner.Name
+						appType = owner.Kind
+						break
+					} else {
+						appName = pod.Name
+						appType = ""
+					}
+				}
```

Comment:
Nothing problematic with your code , but looking at `scaleControllerToZero` seems like we already have some logic to recursively find the root controller . 

For example , nn the case we have this hierarchy `Deployment --> ReplicaSet --> Pod` , this logic will report the `ReplicaSet` which might be fine, not a big issue. But I was wondering if in this case we want the `Deployment` name instead. 

ReplicaSet might be: `host-operator-controller-manager-5d7756db97` while Deployment name is: `host-operator-controller-manager`.

This is already a great improvement, so feel free to leave it as is if it's too complicated to reuse some of that logic for the recursive search . T